src/parser/jobs/brd/modules/Sidewinder.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * @author Ririan
+ */
+import React from 'react'
+import {Trans} from '@lingui/react'
+import {t} from '@lingui/macro'
+import {List} from 'semantic-ui-react'
+import Module from 'parser/core/Module'
+import {getDataBy} from 'data'
+import {TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+import STATUSES from 'data/STATUSES'
+import ACTIONS from 'data/ACTIONS'
+import {ActionLink, StatusLink} from 'components/ui/DbLink'
+
+// All of Bards DoTs
+const DOTS = [
+	STATUSES.CAUSTIC_BITE.id,
+	STATUSES.STORMBITE.id,
+	STATUSES.VENOMOUS_BITE.id,
+	STATUSES.WINDBITE.id,
+]
+
+export default class Sidewinder extends Module {
+	static handle = 'sidewinder'
+	static title = t('brd.sidewinder.title')`Sidewinders and Shadowbites`
+	static dependencies = [
+		'suggestions',
+	]
+
+	_amountOfBadSidewinders = 0
+	_amountOfBadShadowbites = 0
+
+	_badCasts = []
+
+	_enemies = {}
+
+	constructor(...args) {
+		super(...args)
+
+		this.addHook('cast', {
+			by: 'player',
+			abilityId: [ACTIONS.SIDEWINDER.id, ACTIONS.SHADOWBITE.id],
+		}, this._onSidewinderCast)
+
+		this.addHook('applydebuff', {
+			by: 'player',
+			abilityId: DOTS,
+		}, this._onApplyDot)
+
+		this.addHook('removedebuff', {
+			by: 'player',
+			abilityId: DOTS,
+		}, this._onRemoveDot)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onSidewinderCast(event) {
+		const enemyDots = this._getEnemyDots(event.targetID)
+		const abilityId = event.ability.guid
+		if (enemyDots.length < 2) {
+			if (abilityId === ACTIONS.SIDEWINDER.id) {
+				this._amountOfBadSidewinders++
+			} else {
+				this._amountOfBadShadowbites++
+			}
+
+			this._badCasts.push({abilityId: abilityId, dotsApplied: enemyDots.length, appliedDot: enemyDots[0], timestamp: event.timestamp})
+		}
+	}
+
+	_onApplyDot(event) {
```

Comment:
The core `enemies` module tracks statuses already, you can use `enemies.getEntity(<actorId>)` to get an object with info, then if there _is_ info, `enemy.hasStatus` can be used to check for statuses. While not _quite_ the same as what you're tracking, MCH's Wildfire module does something similar if you'd like an example.

Reply:
I didn't know that, that simplifies things a lot.  Thanks!